tests/contract_testcases/substrate/function_types/variable_or_func_type.sol

```diff
@@ -16,6 +16,4 @@ contract bar {
         }
         
 // ---- Expect: diagnostics ----
-// warning: 10:13-33: function can be declared 'view'
-// warning: 12:21-29: local variable 'variable' has been assigned, but never read
-// warning: 14:51-60: local variable 'func_type' has been assigned, but never read
+// error: 14:63-68: function 'total' is overloaded
```

Comment:
This test is no longer working. The test is supposed to check that `total` can be resolved as a variable name, or as a function name (the accessor function).

Note that solc can't do this, this is a solang feature.

Reply:
I was under the impression that the test is wrong. I was confused as to why solang does not detect the same error as `solc` (solc points out the invalid type, which I think is much better). 

> this is a solang feature

How is it used? Feels to me like we should rewrite this test